pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -289,42 +298,57 @@ func decodeIPSecKey(keyRaw string) ([]byte, error) {
 
 // LoadIPSecKeysFile imports IPSec auth and crypt keys from a file. The format
 // is to put a key per line as follows, (auth-algo auth-key enc-algo enc-key)
-func LoadIPSecKeysFile(path string) error {
+func LoadIPSecKeysFile(path string) (uint8, error) {
 	file, err := os.Open(path)
 	if err != nil {
-		return err
+		return 0, err
 	}
 	defer file.Close()
 	return loadIPSecKeys(file)
 }
 
-func loadIPSecKeys(r io.Reader) error {
+func loadIPSecKeys(r io.Reader) (uint8, error) {
+	var spi uint8
+
 	scanner := bufio.NewScanner(r)
 	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
+		offset := 0
+
 		ipSecKey := &ipSecKey{
-			Spi:   1,
 			ReqID: 1,
 		}
 
 		// Scanning IPsec keys formatted as follows,
 		//    auth-algo auth-key enc-algo enc-key
 		s := strings.Split(scanner.Text(), " ")
 		if len(s) < 4 {
-			return fmt.Errorf("missing IPSec keys or invalid format")
+			return 0, fmt.Errorf("missing IPSec keys or invalid format")
 		}
 
-		authkey, err := decodeIPSecKey(s[1])
+		spiI, err := strconv.Atoi(s[0])
 		if err != nil {
-			return fmt.Errorf("unable to decode authkey string %q", s[1])
+			// If no version info is provided assume using key format without
+			// versioning and assign SPI.
+			spiI = 1
+			offset = -1
+		}
+		if spiI > linux_defaults.IPsecMaxKeyVersion {
+			return 0, fmt.Errorf("encryption Key space exhausted, id must be less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion, s[0])
 		}
-		authname := s[0]
+		spi = uint8(spiI)
```

Comment:
I notice that all of your examples start with spi = 1. Is 0 special in any way (eg non-encrypted)? (If special, maybe needs validation, if not, feedback is a no-op here)

Reply:
spi = 0 is no-encryption. Can add a comment and document in documentation.

Reply:
Agree would be nice, been wondering about the same.

Reply:
Is this a valid configuration in the file? Should it be rejected in some way?

Reply:
A configuration like '0 algo foo ...` yeah thats a bit bogus.